### ISSUE TITLE
"Liquid error: Parameter count mismatch." When using | Where filter to compare vs strings with spaces

### ISSUE THREAD

**Issue by NuclearTacos** [SparkDevNetwork/Rock#3943]
### Prerequisites
* [x] Put an X between the brackets on this line if you have done all of the following:
    * Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
    * Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
    * Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack) to see if your bug or enhancement is already reported?

### Description

`| Where` filter doesn't like strings with spaces in them as the filter value.
![image](https://user-images.githubusercontent.com/19762670/66675049-1bc22280-ec2a-11e9-857d-51c41b58638f.png)

### Steps to Reproduce

1. Using stock Config, Create a Connection Request and add Activity to it with an Activity Type that has a space in the Name.
2. Enable Entity Commands in an HTML content block.
3. Paste this Lava in there, but replace the `4` with the Id of the test connection request
```
{% assign activityName = 'Called Left Message' %}
{% connectionrequest id:'4' %}
    {{ connectionrequest | Property:'ConnectionOpportunity.ConnectionType.ConnectionActivityTypes' | Where:'Name','Called Left Message' | First | Property:'Guid' }}<hr>
    {{ connectionrequest | Property:'ConnectionOpportunity.ConnectionType.ConnectionActivityTypes' | Where:'Name',activityName | First | Property:'Guid' }}
{% endconnectionrequest %}
```
( You can drop off filters after the `| Where` filter and experience the same issue )

**Expected behavior:**

I expected that I could compare the value of the Activity's Name properly, regardless of spaces.

**Actual behavior:**

Spaces in the filterValue string seem to cause some issue in how it's evaluated in Liquid or the filter and it doesn't process properly.

### Versions

* **Rock Version:** 9.1
* **Client Culture Setting:** en-us

**Comment by bradcerb** [comment]
Thanks for finding this! I'm having the same problem. Here's my code:

`{{ Workflow | Attribute:'ConnectionRequest','Object' | Property:'ConnectionOpportunity.ConnectionType.ConnectionActivityTypes' | Where:'Name', 'Application Completed' | First | Property:'Guid' }}`

There were spaces around the `WHERE` statement and it was throwing an error: 

> Liquid Error: Parameter Count Mismatch.

I removed everything after the `WHERE` statement and still got the same error.

**Comment by sichiriu** [comment]
Looks like using single quotes around a key containing spaces causes this issue. The filter works properly on 9.2 when using double quotes.

**Comment by allchannelson** [comment]
Adding a space in front of the comma in the Where filter also "fixes" the issue

**Comment by cwomack** [comment]
@NuclearTacos (and others on this thread), there's actually a partial fix that happened in v11.2 to this issue... but v13 should take care of both lines of Lava wth it's new Lava Engine called "Fluid".  Using the sample Lava provided from the top of this issue and breaking down which ones work where/what version: 

`{% assign activityName = 'Called Left Message' %}`
// **With Rock v13 and it's new Lava Engine, this lava gives no errors**
``
{% connectionrequest id:'4' %}
    {{ connectionrequest | Property:'ConnectionOpportunity.ConnectionType.ConnectionActivityTypes' | Where:'Name','Called Left Message' | First | Property:'Guid' }}
``
// **This Lava directly below works in v11.2+** 
``
    {{ connectionrequest | Property:'ConnectionOpportunity.ConnectionType.ConnectionActivityTypes' | Where:'Name',activityName | First | Property:'Guid' }}
``
`{% endconnectionrequest %}`


I'll close this issue out as fixed in v13, but feel free to comment back if there are still issues after Rock v13 is released.
